chore: lint code and text in integration tutorials (#1167)
wandb/docs: content/tutorials/integration-tutorials/huggingface.md

title: Hugging Face
weight: 3
---

{{< img src="/images/tutorials/huggingface.png" alt="" >}}

{{< cta-button colabLink="https://colab.research.google.com/github/wandb/examples/blob/master/colabs/huggingface/Huggingface_wandb.ipynb" >}}

> Visualize your [Hugging Face](https://github.com/huggingface/transformers)

Reviewer: Can you please undo the hard line wrapping that is introduced in this PR? Markdown is easier to maintain and review with soft wrapping per paragraph.

## Install, import, and log in

Install the Hugging Face and Weights & Biases libraries, and the GLUE dataset
and training script for this tutorial.

- [Hugging Face Transformers](https://github.com/huggingface/transformers):
  Natural language models and datasets
- [Weights & Biases]({{< relref "/" >}}): Experiment tracking and visualization
- [GLUE dataset](https://gluebenchmark.com/): A language understanding benchmark
  dataset
- [GLUE script](https://raw.githubusercontent.com/huggingface/transformers/refs/heads/main/examples/pytorch/text-classification/run_glue.py):
  Model training script for sequence classification

```notebook
!pip install datasets wandb evaluate accelerate -qU
!wget https://raw.githubusercontent.com/huggingface/transformers/refs/heads/main/examples/pytorch/text-classification/run_glue.py
```

```notebook
# the run_glue.py script requires transformers dev
!pip install -q git+https://github.com/huggingface/transformers
```

Before continuing,
[sign up for a free account](https://app.wandb.ai/login?signup=true).

> ## Put in your API key

Reviewer: ```suggestion
## Populate your API key
```

Once you've signed up, run the next cell and click on the link to get your API
key and authenticate this notebook.

```python
import wandb
wandb.login()
```

> Optionally, we can set environment variables to customize W&B logging. See
> [documentation]({{< relref "/guides/integrations/huggingface/" >}}).

Reviewer: ```suggestion
To customize W&B logging by setting environment variables, refer to the [Hugging Face documentation]({{< relref "/guides/integrations/huggingface/" >}}).
```

> ```jupyter

Reviewer: ```suggestion
```notebook
```
Also elsewhere in this PR
